internal/cli/serverless/sqluser/create.go

```diff
@@ -0,0 +1,302 @@
+					return errors.Trace(err)
+				}
+				userName = uName
+
+				pw, err := cmd.Flags().GetString(flag.Password)
+				if err != nil {
+					return errors.Trace(err)
+				}
+				password = pw
+
+				uRole, err := cmd.Flags().GetString(flag.UserRole)
+				if err != nil {
+					return errors.Trace(err)
+				}
+				userRole = uRole
+			}
+
+			// generate the built-in role
+			builtinRole, customRoles := GetBuiltinRoleAndCustomRoles(userRole)
+
+			params := iamApi.NewPostV1beta1ClustersClusterIDSQLUsersParams().
+				WithClusterID(clusterID).
+				WithSQLUser(&iamModel.APICreateSQLUserReq{
+					AuthMethod:  util.MYSQLNATIVEPASSWORD,
+					UserName:    userName,
+					Password:    password,
+					BuiltinRole: builtinRole,
+					CustomRoles: customRoles,
+				}).
+				WithContext(ctx)
+
+			_, err = d.CreateSQLUser(params)
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			_, err = fmt.Fprintln(h.IOStreams.Out, color.GreenString("SQL user %s.%s is created", userPrefix, userName))
+			if err != nil {
+				return err
+			}
+			return nil
+
+		},
+	}
+
+	CreateCmd.Flags().StringP(flag.ClusterID, flag.ClusterIDShort, "", "The ID of the cluster.")
+	CreateCmd.Flags().StringP(flag.UserName, "", "", "The name of the SQL user.")
+	CreateCmd.Flags().StringP(flag.Password, "", "", "The password of the SQL user.")
+	CreateCmd.Flags().StringP(flag.UserRole, "", "", "The built-in role of the SQL user.")
+
+	return CreateCmd
+}
+
+// func GetBuiltinRole(userRole string) string {
+// 	role := strings.ToLower(userRole)
+// 	switch role {
+// 	case util.ADMIN:
+// 		role = util.ADMIN_ROLE
+// 	case util.READWRITE:
+// 		role = util.READWRITE_ROLE
+// 	case util.READONLY:
+// 		role = util.READONLY_ROLE
+// 	default:
+// 		role = userRole
+// 	}
+
+// 	return role
+// }
+
+func initialCreateInputModel(userPrefix string) ui.TextInputModel {
+	m := ui.TextInputModel{
+		Inputs: make([]textinput.Model, len(createSQLUserField)),
+	}
+
+	for k, v := range createSQLUserField {
+		t := textinput.New()
+		t.Cursor.Style = config.CursorStyle
+		t.CharLimit = 32
+
+		switch k {
+		case flag.UserName:
+			// add a prefix showing the user prefix
+			t.Placeholder = "User Name"
+			t.Focus()
+			t.PromptStyle = config.FocusedStyle
+			t.TextStyle = config.FocusedStyle
+			t.Prompt = userPrefix + "."
+		case flag.Password:
+			t.Placeholder = "Password"
+			t.EchoMode = textinput.EchoPassword
+			t.EchoCharacter = '•'
+		}
+		m.Inputs[v] = t
+	}
+	return m
+}
+
+func GetUserPrefix(ctx context.Context, d cloud.TiDBCloudClient, clusterID string) (string, error) {
+	params := serverlessApi.NewServerlessServiceGetClusterParams().WithClusterID(clusterID).WithContext(ctx)
+
+	cluster, err := d.GetCluster(params)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+
+	return cluster.Payload.UserPrefix, nil
+}
+
+func GetBuiltinRoleAndCustomRoles(userRole string) (string, []string) {
+	lowerRole := strings.ToLower(userRole)
+	switch lowerRole {
+	case util.ADMIN:
```

Comment:
when in interactive mode, shouldn't this be `util.ADMIN_DISPLAY`?

Reply:
> when in interactive mode, shouldn't this be `util.ADMIN_DISPLAY`?

The return value of this function is for API

Reply:
> And what if in non-interactive mode, the user passed in `role_admin, custom1, custom2`?

Yes, and I think there will be a bug that when user inputs `admin, custom1, custom2`, the `admin` should be converted to `role_admin`